icgcget/clients/ega/ega_client.py

```diff
@@ -59,3 +61,23 @@ def ega_call(object_id, username, password, tool_path, parallel, udt, download_d
     if rc_decrypt != 0:
         return rc_decrypt
     return 0
+
+
+def ega_access_check(username, password):
+    request = "https://ega.ebi.ac.uk/ega/rest/access/v2/users/" + quote(username) + "?pass=" + quote(password)
+    try:
+        resp = call_api(request, "https://ega.ebi.ac.uk/ega/rest/")
+    except HTTPError:
+        return False
+    if resp["header"]["userMessage"] == "OK":
+        session_id = resp["response"]["result"][1]
+        request2 = "https://ega.ebi.ac.uk/ega/rest/access/v2/datasets?session=" + session_id
+        try:
+            resp2 = call_api(request2, "https://ega.ebi.ac.uk/ega/rest/")
+            data_sets = resp2["response"]["result"]
+        except HTTPError:
+            return False
+        if "EGAD00001000023" in data_sets and "EGAD00010000562" in data_sets:
```

Comment:
What's this? Please add comments to non-obvious parts of the code. Its not apparent what your intent is here. Naming variables go a long way.